### PR TITLE
Non-Integer inputs

### PR DESCRIPTION
**Pull request by sammy00747**
Changes have been made in inc/helper.h and src/main.c
In this pull request, non-integer inputs have been taken care..
Here, position input is taken as string..
And if there are any non-digit character in the string (also if the string 
has more than 2 or no character) then the input is not accepted.
Else the input is accepted and converted to corresponding integer..
Hope it helps..

### CODE REVIEW

File: src/main.c [hunk 1]
```diff
@@ -25,18 +25,57 @@ int main(int argc, char* argv[])
 		return 1;
 	}
 
-	int result = 0, pos, player = 0;
+	int result = 0, pos, player = 0,i,sw=0;
```
Comment (RudraNilBasu) [inline src/main.c]: Use a more meaningful name instead of `sw` (same for most other newly created variables)

File: src/main.c [hunk 2]
```diff
@@ -25,18 +25,57 @@ int main(int argc, char* argv[])
...
 	init(N, Board);
 	greet();
 
 	while (show(N, Board), !result) {
 		do {
 			printf("\nFor Player(\'%c\'):\n", Tic[player]);
-			printf("Enter Position: "); 
-			assert(scanf("%d", &pos) == 1);
+			printf("Enter Position (non-digit characters not allowed.. not even white spaces...): ");
+			goto init__1;
```
Comment (RudraNilBasu) [inline src/main.c]: Using `goto` is a bad practice. Still to be confirmed with @Progyan1997

File: src/main.c [hunk 3]
```diff
@@ -25,18 +25,57 @@ int main(int argc, char* argv[])
...
+			free(s);
+			init__1:
+			i=0;
+			s=malloc(1);
+			do
+			{
+				c = getchar();
+				s[i++] = c;
+				s = realloc(s,i+1);
+			}while(c != '\n' && c != EOF);
```
Comment (RudraNilBasu) [inline src/main.c]: Follow the coding style: https://bytesclub.github.io/docs/coding_style.pdf

File: src/main.c [hunk 4]
```diff
@@ -25,18 +25,57 @@ int main(int argc, char* argv[])
...
+			s[i] = '\0';
+			switch(strlen(s))
+			{
+				case 2:
+				if(!(isdigit(s[0])))
+				{
+					printf(ANSI_YELLOW "Wrong input.. Enter again...\n" ANSI_RESET);
+					goto error_pos;
+				}
+				else
```
Comment (RudraNilBasu) [inline src/main.c]: Same here.

**Review by RudraNilBasu — CHANGES_REQUESTED** [review]
See inline comments

File: src/main.c (hunk 2 above)
Comment (0xTheProDev) [reply src/main.c]: We're strictly against of use of `goto`. Try alternatives if possible. @sammy00747

Comment (sammy00747) [reply src/main.c]: not possible.. atleast, i don't know..

File: src/main.c (hunk 1 above)
Comment (sammy00747) [reply src/main.c]: naam di66i avoid_error..

**Pull request closed without merging**